packages/design-system/src/components/datalist/datalist.js

```diff
@@ -202,7 +204,12 @@ const Datalist = forwardRef(function Datalist(
       />
       {isOpen && !disabled && isInline && list}
       {!disabled && !isInline && (
-        <Popup anchor={internalRef} isOpen={isOpen} fillWidth={DEFAULT_WIDTH}>
+        <Popup
+          anchor={internalRef}
+          isOpen={isOpen}
+          fillWidth={DEFAULT_WIDTH}
+          zIndex={zIndex}
```

Comment:
z index needs overriding here because by default (2) it's not high enough to be on top of the story details modal